Improve documentation after defining API (#12)
kyma-incubator/octopus: README.md

## Overview

Octopus is a testing framework that allows you to run tests defined as Docker images on a running cluster.

> It was created to replace the `helm test` with providing additional features:

Reviewer: ```suggestion
It was created to replace the `helm test` by providing these additional features:
```

> 1. Selective testing.  

Reviewer: ```suggestion
- Selective testing, which is especially useful for local development when you want to execute only tests that local changes can affect.
```

This can be especially useful for local development when a developer wants to execute only tests that 
his local changes can affect.  

> 1. Retries on failed tests. 

Reviewer: ```suggestion
- Retries on failed tests, which aims to improve tests and make them stable.
```

Author: To discuss

Flaky tests are an everyday reality. Definitely, developers should strive to improve tests and make them stable, but we should accept that they will exist.

> 1. Run tests multiple times.

Reviewer: ```suggestion
- Running tests multiple times, which is helpful when you add a new test and you want to validate if it is stable, or when you want to reproduce a problem that occurs on CI.
```

This can be helpful when a developer adds a new test and he/she needs to validate if a test is stable or he wants to reproduce a problem that sometimes occurs on CI.

> 1. Full support for concurrent testing. 

Reviewer: ```suggestion
- Full support for concurrent testing, which allows you to run a given test concurrently with other tests.
```

Author: I changed it a little bit, let me know what you think

Octopus is a testing framework that allows you to run tests defined as Docker images on a running cluster.

> ## Purpose

Reviewer: I'd remove this section and merge it with the overview. Maybe you can start with such a sentence:

``It was created to supplement the `helm test` with the following features:``

... or something like that. Then you can list all the reasons why Octopus is so awesome. I'd focus rather on Octopus advantages rather than helm test limitations.
